### ISSUE TITLE
Waiting the end of a matrix run 

### ISSUE THREAD

**Issue by tristanlatr** [lewagon/wait-on-check-action#4]
Hello

I have a workflow file like

```yml
name: test
on: 
  push:
    branches: 
      - master
jobs: 
  test:
    name: ${{ matrix.os }} py${{ matrix.python-version }}
    runs-on: ${{ matrix.os }} 

    strategy:
      matrix:
        os: [ubuntu-latest, windows-latest, macos-latest]
        python-version: ['3.6','3.7','3.8','3.9']
      max-parallel: 1
      fail-fast: false

    steps:
      - uses: actions/checkout@v2
      - name: Set up Python ${{ matrix.python-version }}
        uses: actions/setup-python@v2
        with:
          python-version: ${{ matrix.python-version }}

      - name: Install dependencies
        run: |
          python -m pip install --upgrade pip setuptools wheel
          python -m pip install -r requirements.txt
          python -m pip install .

      - name: Run tests
        run: python -m pytest
```

The issue is that the job names are automatically generated from the OS and python version...
How to wait until all those tests are run ?

Thanks

**Comment by progapandist** [comment]
Hi! Thank you for using our little action. The only solution I see at the moment is hardcoding those job names, exactly as you did in a referenced commit. The job name has to be known, as an external call to Checks API that drives this action requires it :(

**Comment by Billy-** [comment]
Hi, firstly, thanks for this action!

I'd also like to be able to wait for all tests to pass. In my case, when a PR is merged we run our release, which in turn pushes back a release commit to our master branch, which doesn't run any jobs at all. We have an auto-approve action for automatic update PRs, which then get auto-merged. I'd like the auto-approve action to wait if master is currently running any jobs (meaning it's in the middle of a release). If I use this action as it currently is, telling it to wait for the `release` job, I'm assuming it'll fail once the release commit comes back as that job won't run.

I also think waiting for all checks to pass would be a useful feature for many other use cases. Even if just a convenience over having many wait-on-check jobs, each one for a separate job?

I could have a go at implementing this if you like. I'm assuming the GitHub checks API allows us to query what checks are running/expected. So we can use that. Then for the action - could we allow omitting the `check-name` in the config and defaulting to "all"? This would technically be a non-breaking change since omitting this propert was not supported previously

**Comment by Billy-** [comment]
I just kinda realised this doesn't necessarily totally solve the original issue here. Only because it allows waiting for _all_ checks but not _some_ checks.

I would guess the best approach for that would be to allow passing some sort of glob/regex e.g `checkName: "my-job-*-py*"` and/or allowing an array of checks being passed?

**Comment by progapandist** [comment]
I would have to play with matrix runs more to give some valuable input here. None of my workflows require them ATM. Can you give me an example of an open repo that I can fork and try?

**Comment by Billy-** [comment]
I have never played with matrix runs either.. I should probably have opened another issue for the PR I created (and still can if you like) as I'm not sure it specifically addresses this issue.. although it may help.

My PR simply waits for all checks, which may be fine if the matrix is the only thing running checks, but of course if there are other checks then it'd wait for those too. Also assuming that the matrix runs just produce separate checks in the checks API - which based on https://github.com/mfesiem/msiempy/commit/2b6f6e5b4f87300ad9d71ff1fa258e8a86777fce already using this action, just stating each check separately, I guess it would work.

I think my PR is still useful as a separate feature, but to address this issue better we could add the option to pass a glob or regex or something to be able to select jobs that way?

**Comment by matiasalbarello** [comment]
@tristanlatr, does the current implementation with regular expressions solve this issue?

**Comment by jeanlescure** [comment]
Since this is a top google result, here's a workaround I used for my case:

```yaml
name: tests
on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main
jobs:
  tests1:
    name: Test on Node ${{ matrix.node }} and ${{ matrix.os }}
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        node: ['14.x']
        os: [ubuntu-latest]

  # ...

  tests2:
    name: Test on Node ${{ matrix.node }} and ${{ matrix.os }}
    needs: tests1
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        node: ['16.x']
        os: [ubuntu-latest]

  # ...

  tests3:
    name: Test on Node ${{ matrix.node }} and ${{ matrix.os }}
    needs: tests2
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        node: ['14.x']
        os: [macOS-latest]

  # ...

  tests4:
    name: Test on Node ${{ matrix.node }} and ${{ matrix.os }}
    needs: tests3
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        node: ['16.x']
        os: [macOS-latest]

  # ...

```

I needed to run the tests one after another due to a rate limit that, even though was taken into account at code level, when running 4 jobs in parallel it would easily hit the rate limit due to sending multiple requests at the same time.

With this workaround I even got this nice alternate view of the tests 😄 

![2022-01-10_21-03](https://user-images.githubusercontent.com/3330339/148874504-ba0b85eb-f4d0-4625-946e-2c92518ad59f.png)
